modules/gdscript/gdscript.cpp

```diff
@@ -2001,114 +2001,116 @@ String GDScriptLanguage::get_global_class_name(const String &p_path, String *r_b
+			msg = "The method '" + symbols[0] + "()' was not found in base '" + symbols[1] + "' but there's a constant with the same name. Did you mean to access it?";
 		} break;
 		case FUNCTION_USED_AS_PROPERTY: {
 			CHECK_SYMBOLS(2);
-			return "The property '" + symbols[0] + "' was not found in base '" + symbols[1] + "' but there's a method with the same name. Did you mean to call it?";
+			msg = "The property '" + symbols[0] + "' was not found in base '" + symbols[1] + "' but there's a method with the same name. Did you mean to call it?";
 		} break;
 		case INTEGER_DIVISION: {
-			return "Integer division, decimal part will be discarded.";
+			msg = "Integer division, decimal part will be discarded.";
 		} break;
 		case UNSAFE_PROPERTY_ACCESS: {
 			CHECK_SYMBOLS(2);
-			return "The property '" + symbols[0] + "' is not present on the inferred type '" + symbols[1] + "' (but may be present on a subtype).";
+			msg = "The property '" + symbols[0] + "' is not present on the inferred type '" + symbols[1] + "' (but may be present on a subtype).";
 		} break;
 		case UNSAFE_METHOD_ACCESS: {
 			CHECK_SYMBOLS(2);
-			return "The method '" + symbols[0] + "' is not present on the inferred type '" + symbols[1] + "' (but may be present on a subtype).";
+			msg = "The method '" + symbols[0] + "' is not present on the inferred type '" + symbols[1] + "' (but may be present on a subtype).";
 		} break;
 		case UNSAFE_CAST: {
 			CHECK_SYMBOLS(1);
-			return "The value is cast to '" + symbols[0] + "' but has an unknown type.";
+			msg = "The value is cast to '" + symbols[0] + "' but has an unknown type.";
 		} break;
 		case UNSAFE_CALL_ARGUMENT: {
 			CHECK_SYMBOLS(4);
-			return "The argument '" + symbols[0] + "' of the function '" + symbols[1] + "' requires a the subtype '" + symbols[2] + "' but the supertype '" + symbols[3] + "' was provided";
+			msg = "The argument '" + symbols[0] + "' of the function '" + symbols[1] + "' requires a the subtype '" + symbols[2] + "' but the supertype '" + symbols[3] + "' was provided";
 		} break;
 		case DEPRECATED_KEYWORD: {
 			CHECK_SYMBOLS(2);
-			return "The '" + symbols[0] + "' keyword is deprecated and will be removed in a future release, please replace its uses by '" + symbols[1] + "'.";
+			msg = "The '" + symbols[0] + "' keyword is deprecated and will be removed in a future release, please replace its uses by '" + symbols[1] + "'.";
 		} break;
 		case STANDALONE_TERNARY: {
-			return "Standalone ternary conditional operator: the return value is being discarded.";
-		}
+			msg = "Standalone ternary conditional operator: the return value is being discarded.";
+		} break;
 		case WARNING_MAX:
-			break; // Can't happen, but silences warning
+			ERR_FAIL_V_MSG(String(), "Invalid GDScript warning code: " + get_name_from_code(code) + ".");
 	}
-	ERR_FAIL_V_MSG(String(), "Invalid GDScript warning code: " + get_name_from_code(code) + ".");
+    return msg + " (" + get_name() + ")";
```

Comment:
Indentation issue.

I'd suggest using square brackets (`[UNUSED_ARGUMENT]`) instead of parentheses, as it's what is used for compiler warnings (at least GCC and Clang's).

Reply:
I'm fine either way, but FWIW pylint uses parens, and gdscript seems most similar to python:

```
[rcorre@midova tmp]$ pylint example.py
************* Module example
example.py:1:0: C0305: Trailing newlines (trailing-newlines)
```

Anyone familiar with python will immediately understand that "the thing in parens" is what they can put in the ignore comment.

Reply:
Alright, I'm also fine either way :) I guess it depends what compiler warnings folk are most used too (C/C++ vs Python).